Lib/test/test_importlib/source/test_file_loader.py

```diff
@@ -206,6 +206,8 @@ def test_timestamp_overflow(self):
             except OverflowError:
                 self.skipTest("cannot set modification time to large integer")
             except OSError as e:
+                if sys.platform == "os400" and e.errno == getattr(errno, 'EINVAL', None):
+                    self.skipTest("On IBM i IFS doesn't support times past 2038")
                 if e.errno != getattr(errno, 'EOVERFLOW', None):
```

Comment:
```suggestion
                errnos = [getattr(errno, 'EOVERFLOW', None)]
                if sys.platform == 'os400':
                    errnos.append(getattr(errno, 'EINVAL', None))
                if e.errno not in errnos:
```

We can combine this with the original check.